recipes/correct-ts-specifiers/src/logger.ts

```diff
@@ -0,0 +1,46 @@
+import process from 'node:process';
+
+type LogMsg = string;
+type LogType = 'error' | 'log' | 'warn';
+type FileLog = { msg: LogMsg; type: LogType };
+type Source = URL['pathname'];
+
+/**
+ * Collect log entries and report them at the end, collated by source module.
+ */
+export const logger = (source: Source, type: LogType, msg: LogMsg) => {
+	const fileLog = new Set<FileLog>(logs.has(source) ? logs.get(source) : []);
+
+	fileLog.add({ msg, type });
+	logs.set(source, fileLog);
+};
+
+/**
+ * @type {Map<URL['pathname'], Set<Record<LogMsg, LogType>>>}
+ */
+const logs = new Map<Source, Set<FileLog>>();
+
+process.once('beforeExit', emitLogs);
+
+/* node:coverage disable */
+// this is actually coveraged, but v8 doesn't recognise it
+function emitLogs() {
+	let hasError = false;
+
+	for (const [sourceFile, fileLog] of logs.entries()) {
+		console.log('[Codemod: correct-ts-specifiers]:', sourceFile);
```

Comment:
maybe use `styleText` to have something coooollll

Reply:
MVP first 😜 been working on this thing since like June

Reply:
Oh yeah, holly molly macaroni